### PR TITLE
Medical updates v2

### PR DESCRIPTION
**Pull request by LiekeSchoenmaker**
Different updates requestered from medical team

### CODE REVIEW

File: patient-reporter/src/main/java/com/hartwig/oncoact/patientreporter/lama/LamaInterpretation.java [hunk 1]
```diff
@@ -36,12 +37,23 @@ public static String hospitalContactReport(@NotNull PatientReporterData lamaPati
...
         } else {
-            LOGGER.warn("None requester name of report is known. Solve before reporting!");
+            LOGGER.info(requester);
+            LOGGER.info(studyPI);
+            if (requester != null) {
+                requesterNameReport = requester;
+            } else if (studyPI != null) {
+                requesterNameReport = studyPI;
+            } else {
+                LOGGER.warn("None requester name of report is known. Solve before reporting!");
```
Comment (mvanniekerkHartwig) [inline patient-reporter/src/main/java/com/hartwig/oncoact/patientreporter/lama/LamaInterpretation.java:55]: Same here.

**Review by mvanniekerkHartwig — COMMENTED** [review]
Looks good, I only have one question and one suggestion.

File: patient-reporter/src/main/java/com/hartwig/oncoact/patientreporter/lama/LamaInterpretation.java [hunk 2]
```diff
@@ -36,12 +37,23 @@ public static String hospitalContactReport(@NotNull PatientReporterData lamaPati
...
-        if (studyPI != null) {
-            requesterNameReport = studyPI;
-        } else if (requester != null) {
-            requesterNameReport = requester;
+
+        if (lamaPatientData.getIsStudy()) {
+            if (studyPI != null) {
+                requesterNameReport = studyPI;
+            } else {
+                LOGGER.warn("None requester name of report is known. Solve before reporting!");
```
[the remark this answers is not in the document]
Comment (LiekeSchoenmaker) [reply patient-reporter/src/main/java/com/hartwig/oncoact/patientreporter/lama/LamaInterpretation.java]: both are fine I think. But have implement a try/catch function.

File: patient-reporter/src/main/java/com/hartwig/oncoact/patientreporter/lama/LamaInterpretation.java [hunk 3]
```diff
@@ -36,12 +37,25 @@ public static String hospitalContactReport(@NotNull PatientReporterData lamaPati
...
-            requesterNameReport = studyPI;
-        } else if (requester != null) {
-            requesterNameReport = requester;
+
+        if (lamaPatientData.getIsStudy()) {
+            try {
+                if (studyPI != null) {
+                    requesterNameReport = studyPI;
+                }
+            } catch (NullPointerException exception) {
```
Comment (mvanniekerkHartwig) [inline patient-reporter/src/main/java/com/hartwig/oncoact/patientreporter/lama/LamaInterpretation.java:46]: This last change makes no sense, there is nothing to catch here. I'd rather you revert this change.
I meant you could change the code to something like:
```java
            if (studyPI != null) {
                requesterNameReport = studyPI;
            } else {
                throw new IllegalArgumentException("None requester name of report is known. Solve before reporting!");
            }
```

But only if you're ok with stopping a patient reporter run prematurely on missing this information (which I think is probably justified if it needs to be fixed before reporting anyway).

**Review by mvanniekerkHartwig — CHANGES_REQUESTED** [review]

**Review by mvanniekerkHartwig — APPROVED** [review]

**Pull request merged**